server.go

```diff
@@ -1,11 +1,25 @@
 package api2
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"reflect"
 )
 
+type errorMessage struct {
+	Error         string          `json:"error"`
+	OrigError     json.RawMessage `json:"orig_error,omitempty"`
```

Comment:
I do not like this name. It wounds really weird. As public api I'd prefer using code field.

Reply:
renamed "orig_error" to "detail"
renamed "orig_error_type" to "code"